src/cdomains/floatOps/floatOps.ml

```diff
@@ -0,0 +1,61 @@
+type round_mode =
+  | Nearest
+  | ToZero
+  | Up
+  | Down
+
+module type CFloatType = sig
+  type t
+
+  val zero: t
+  val max_value: t
+  val min_value: t
+
+  val equal: t -> t -> bool
+  val compare: t -> t -> int
+  val to_yojson: t -> Yojson.Safe.t
+
+  val to_string: t -> string
+  val is_finite: t -> bool
+
+  val add: round_mode -> t -> t -> t
+  val sub: round_mode -> t -> t -> t
+  val mul: round_mode -> t -> t -> t
+  val div: round_mode -> t -> t -> t
+  val atof: round_mode -> string -> t
+end
+
+module CDouble = struct
+  type t = float [@@deriving eq, ord, to_yojson]
+
+  let to_string = Float.to_string
+  let is_finite = Float.is_finite
+  let zero = 0.
+  let max_value = Float.max_float
+  let min_value = -. Float.max_float
```

Comment:
I would be careful with `min_value`. Usually, when one talks about  "min value" for floats, they are talking about the least representable number greater than zero (see `DBL_MIN` for C or `Float.min_float` for OCaml). 
Also whenever someone actually needs the smallest (negative) value for a floating-point number they can just `-. max_value` since in contrast to integers the smallest and largest value have the same absolute value

Reply:
I see your point and, personally, i dislike this naming in general (`smallest_float` would be a way better name for what is now called `min_float`). And for integer values the naming is `min_int` actually denotes the "highest" negative value. Also, getting the respective negative extreme value is not that easy (due to some ocaml type things...) and would look like this `Float_t.sub Nearest Float_t.zero Float_t.max_value`. I don't know, what is the best in this situation? Do you prefer only having a `max_value` and computing the respective min or do you have a alternative naming idea?

Reply:
I myself was very confused when I found out about "min float" being the smallest value greater than zero. Intuitively for me was "min float = - max float" as well.
I think maybe just a different name for it would work? like "neg_max_value" for example.
"`Float_t.neg_max_value`" might be nicer than "`Float_t.sub Nearest Float_t.zero Float_t.max_value`" ?
But I am not sore about the best action here either...

Reply:
The problem with `Float_t.neg_max_value` , at least how i see it, is strictly speaking `- Float.float_min`. What about `Float_t.upper_bound` and `Float_t.lower_bound` ?

Reply:
Upper bound/lower bound sounds fitting and has no conflict with anything I think